### ISSUE TITLE
Guidance on what counts as a non-user visable change

### ISSUE THREAD

**Issue by oxinabox** [SciML/ColPrac#11]
I think it might be useful to define what a non-user visible change is.
Possibly with reference to making releases.
In a sense perhaps a non-user visible change is a 3rd category, like bug-fix or feature.

When I say non-user-visible change I mean one of:
- Refactoring to change internal structure (but not the public API)
- Updating a comment
- renaming a variable
- Fixing a mistake in the tests

Some things sound similar like:
 - Performance improvements
 - updating documentation.

But are not.

Was reminded of this in
https://github.com/JuliaDiff/ChainRules.jl/pull/296#issuecomment-716502527

**Comment by ChrisRackauckas** [comment]
Yes, good point. I'd also add a big one: changes to arguments in non-public APIs. It's really the definition of non-public: you can break it at will, and it's not a breaking change.

**Comment by oxinabox** [comment]
Yep, basically a special case of refactoring.
